src/main/java/io/github/syst3ms/skriptparser/lang/entries/SectionConfiguration.java

```diff
@@ -0,0 +1,125 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SectionConfiguration {
+	@Nullable
+	private CodeSection parent;
+	private final List<EntryLoader> entries = new ArrayList<>();
+	private final Map<String, Object> data = new HashMap<>();
+
+	public SectionConfiguration addOption(String key) {
+		entries.add(new OptionLoader(false, key, false));
+		return this;
+	}
+
+	public SectionConfiguration addLiteral(String key, Class<?> typeClass) {
+		entries.add(new LiteralLoader<>(typeClass, false, key, false));
+		return this;
+	}
+
+	public SectionConfiguration addOptionList(String key) {
+		entries.add(new OptionLoader(true, key, false));
+		return this;
+	}
+
+	public SectionConfiguration addLiteralList(String key, Class<?> typeClass) {
+		entries.add(new LiteralLoader<>(typeClass, true, key, false));
+		return this;
+	}
+
+	public SectionConfiguration addSection(String key) {
+		entries.add(new SectionLoader(key, false));
+		return this;
+	}
+
+	public SectionConfiguration addLoader(EntryLoader loader) {
+		entries.add(loader);
+		return this;
+	}
+
+	public boolean loadConfiguration(@Nullable CodeSection parent, FileSection section, ParserState parserState, SkriptLogger logger) {
+		boolean successful = true;
+		this.parent = parent;
+
+		outer:
+		for (var entry : entries) {
+			for (var el : section.getElements()) {
+				logger.setLine(el.getLine() - 1);
+
+				if (logger.hasError()) {
+					/*
+					 * If the execution of 'loadEntry' caused errors, it means that we
+					 * should not continue parsing the other sections, as specified in
+					 * the Javadoc.
+					 * We finalize the logs and move on to the next entry.
+					 */
+					logger.finalizeLogs();
+					successful = false;
+					continue outer;
+				}
+				if (el instanceof VoidElement)
+					continue;
+				if (entry.loadEntry(this, el, parserState, logger))
+					continue outer;
+			}
+			if (entry.isOptional())
+				continue;
+			// If we're here, it means no value matched and the entry hasn't been configured.
+			// Only the section line is relevant.
+			logger.setLine(section.getLine() - 1);
+			logger.error("The entry named '" + entry.key + "' has not been configured", ErrorType.SEMANTIC_ERROR);
+			logger.finalizeLogs();
+			successful = false;
+		}
+
+		if (!successful) {
+			// Add a final error message stating the section has not been configured correctly.
+			// Only the section line is relevant.
+			logger.setLine(section.getLine() - 1);
+			logger.error("The section '" + section.getLineContent() + "' has not been configured correctly", ErrorType.SEMANTIC_ERROR);
+		}
+		return successful;
+	}
+
+	@Nullable
+	public CodeSection getParent() {
+		return parent;
+	}
+
+	public Map<String, Object> getData() {
+		return data;
+	}
+
+	public Object getValue(String key) {
+		return data.get(key);
+	}
+
+	@SuppressWarnings("unchecked")
+	public <T> T getValue(String key, Class<T> cls) {
```

Comment:
When I had 
```
key: true
```
and used
```java
configuration.getValue("key", Boolean.class);
```
The returned value was always a String, so this would cause a class cast exception.
```
java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
```

Reply:
This is indeed confusing.

Reply:
Added a descriptive error message containing information about this special use-case, but honestly, I'm not sure if I will keep adding these 'edge-case error messages'.